component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -165,8 +165,16 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                 // if no IDPs were found for a given name, the IdentityProviderManager returns a dummy IDP with the
                 // name "default". We need to handle this case.
                 if (StringUtils.equalsIgnoreCase(identityProvider.getIdentityProviderName(), DEFAULT_IDP_NAME)) {
-                    log.error("Default IDP Found. This means a valid IDP was not found for issuer: " + jwtIssuer);
-                    handleException("No Registered IDP found for the JWT with issuer name : " + jwtIssuer);
+                    //the protocol type should be passed to getIdPByAuthenticatorPropertyValue() inorder to get the
+                    // authenticator properly
+
+                    //if id token is used as the assertion below code snippet will be executed
+                    identityProvider = IdentityProviderManager.getInstance().
+                            getIdPByAuthenticatorPropertyValue("IdPEntityId", jwtIssuer
+                                    , tenantDomain, false);
+                    if (identityProvider == null) {
```

Comment:
This will not be null under AFAIK since IdentityProviderManager will return an IDP with name 'default' in the worst case if it cannot find any other IDPs matching the criteria. Better to check the code.